### ISSUE TITLE
LDDTool is throwing ERROR SetMasterAttrXMLBaseDataTypeFromDataType - Data Type is missing - 

### ISSUE THREAD

**Issue by jshughes** [NASA-PDS/pds4-information-model#733]
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When LDDTool was run, it threw "ERROR SetMasterAttrXMLBaseDataTypeFromDataType - Data Type is missing".

@jordanpadams this error was addressed by #724  IMTool Refactoring - Deprecate the Java "Deprecated Object Array". Merged  #725. 

I am not able replicate the error locally using the staged version of the IM #727, merged two weeks ago. But am I am probably missing something. 

Not sure that it would help, but could you provide a link to PDS4_KPLO_IngestLDD.xml

### 🕵️ Expected behavior

I expected no errors.

### 📜 To Reproduce

```bash
1.
2.
3.
...
```

### 🖥 Environment Info

```bash
- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...
```

### 🦄 Related requirements

🦄 #xyz

**Comment by jordanpadams** [comment]
@jshughes here is the KPLO LDD: https://github.com/pds-data-dictionaries/ldd-kplo/blob/main/src/PDS4_KPLO_IngestLDD.xml

In the future, all IngestLDDs are posted on GitHub here: https://github.com/pds-data-dictionaries/

**Comment by jshughes** [comment]
@jordanpadams I found two problems and am working on them. Also, I will need to sort through a few things to understand why the LDDTool run in my local Maven build did not catch the problems.

Question: You mentioned adding some smoke tests to the repo. Just wondering how the smoke tests are different from the Maven build test. It is simply a LDDTool run against an LDD.

**Comment by jordanpadams** [comment]
@jshughes the maven tests always run, but I don’t think we have too many junit tests running? Regardless, the smoke tests take the built lddtool package and run it against a couple dLDDs to make sure they don’t break anything. See https://github.com/NASA-PDS/pds4-information-model/pull/735

**Comment by miguelp1986** [comment]
> @jshughes here is the KPLO LDD: https://github.com/pds-data-dictionaries/ldd-kplo/blob/main/src/PDS4_KPLO_IngestLDD.xml
> 
> In the future, all IngestLDDs are posted on GitHub here: https://github.com/pds-data-dictionaries/

@jordanpadams confirming that the above file is a test example used to ensure that we do not see the error from this issue.

**Comment by jordanpadams** [comment]
@miguelp1986 yes that data works. You should also be able to test with just about any of the LDDs from https://github.com/pds-data-dictionaries/
